### PR TITLE
Database updating

### PR DESCRIPTION
**Pull request by DenisAlicic**
# Database update 

Deleting whole table and populating with data from txt files.

### CODE REVIEW

**Comment by DenisAlicic** [comment]
@mrdakj Please take a look

File: TennisAssociation/Controllers/UpdateDataController.cs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+                        ? (short?) short.Parse(row[columnNames["bestRankingDouble"]])
+                        : null;
+                    player.BestRankingSingle = row[columnNames["bestRankingSingle"]] != null
+                        ? (short?) short.Parse(row[columnNames["bestRankingSingle"]])
+                        : null;
+                    player.Img = Convert.FromBase64String(row[columnNames["img"]]);
+                    return player;
+                });
+            service.PrepareData();
+            service.UpdateData(db.Players, "Players");
```
Comment (mrdakj) [inline TennisAssociation/Controllers/UpdateDataController.cs]: Can you return  service.PrepareData() && service.UpdateData(db.Players, "Players"); as above?

File: TennisAssociation/Services/DatabaseUpdaterService.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            _db.Database.ExecuteSqlRaw("TRUNCATE TABLE " + tableName);    
+                        
+            foreach (var t in _itemsForAdding)
+            {
+                changingSet.Add(t);
+            }
+
+            _db.SaveChanges();
+
+            return true;;
```
Comment (mrdakj) [inline TennisAssociation/Services/DatabaseUpdaterService.cs]: minor: extra ;

File: TennisAssociation/Services/DatabaseUpdaterService.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            }
+            
+            foreach (var row in rows)
+            {
+                T t = _builder.Invoke(row, columnNumbers); 
+                
+                _itemsForAdding.Add(t);
+            }
+
+
```
Comment (mrdakj) [inline TennisAssociation/Services/DatabaseUpdaterService.cs:50]: minor: extra new line

File: TennisAssociation/Services/DatabaseUpdaterService.cs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        public DatabaseUpdaterService(string scriptDirectory, string scriptName, string pathToFile, TennisAssociationContext context, Func<List<string>, Dictionary<string, int >, T> builder)
+        {
+            this._pathToFile = pathToFile;
+            this._scriptName = scriptName;
+            _reader = new CsvReader(pathToFile);
+            _scriptDirectory = scriptDirectory; 
+            _db = context;
+            _builder = builder;
+            _itemsForAdding = new List<T>();
+        }
```
Comment (mrdakj) [inline TennisAssociation/Services/DatabaseUpdaterService.cs:33]: minor: add a new line

File: TennisAssociation/Controllers/UpdateDataController.cs [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+                        ? (int?) int.Parse(row[columnNumbers["semifinal points"]])
+                        : null,
+                    SemifinalPrize = row[columnNumbers["semifinal prize"]] != null
+                        ? (int?) int.Parse(row[columnNumbers["semifinal prize"]])
+                        : null,
+                    FinalPoints = row[columnNumbers["final points"]] != null
+                        ? (int?) int.Parse(row[columnNumbers["final points"]])
+                        : null,
+                    FinalPrize = row[columnNumbers["semifinal prize"]] != null
+                        ? (int?) int.Parse(row[columnNumbers["final prize"]])
```
Comment (mrdakj) [inline TennisAssociation/Controllers/UpdateDataController.cs:76]: semifinal

File: TennisAssociation/Controllers/UpdateDataController.cs [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+                        : null,
+                    WinnerPoints = row[columnNumbers["winner points"]] != null
+                        ? (int?) int.Parse(row[columnNumbers["final points"]])
```
Comment (mrdakj) [inline TennisAssociation/Controllers/UpdateDataController.cs]: winner

File: TennisAssociation/Controllers/UpdateDataController.cs [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+                    SemifinalPoints = row[columnNumbers["semifinal points"]] != null
+                        ? (int?) int.Parse(row[columnNumbers["semifinal points"]])
+                        : null,
+                    SemifinalPrize = row[columnNumbers["semifinal prize"]] != null
+                        ? (int?) int.Parse(row[columnNumbers["semifinal prize"]])
+                        : null,
+                    FinalPoints = row[columnNumbers["final points"]] != null
+                        ? (int?) int.Parse(row[columnNumbers["final points"]])
+                        : null,
+                    FinalPrize = row[columnNumbers["semifinal prize"]] != null
```
Comment (mrdakj) [inline TennisAssociation/Controllers/UpdateDataController.cs]: final

File: TennisAssociation/Controllers/UpdateDataController.cs [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+        }
+
+        [HttpGet("tournaments")]
+        // GET
+        public bool UpdateTournaments()
+        {
+            IDatabaseUpdater<Tournament> service = new DatabaseUpdaterService<Tournament>(ScriptDirectory, TournamentScript, TournamentsFile, db,
+                (row, columnNumbers) => new Tournament
+                {
+                    _1RoundPrize = row[columnNumbers["1. round prize"]] != null
```
Comment (mrdakj) [inline TennisAssociation/Controllers/UpdateDataController.cs:36]: minor: can you sort columns as in txt

File: TennisAssociation/Controllers/UpdateDataController.cs (hunk 1 above)
Comment (DenisAlicic) [reply TennisAssociation/Controllers/UpdateDataController.cs]: I will. :)

**Pull request merged**